hazelcast-jet-hadoop/src/main/java/com/hazelcast/jet/connector/hadoop/WritableSerializerHook.java

```diff
@@ -25,56 +24,58 @@
 import org.apache.hadoop.io.Writable;
 
 import java.io.IOException;
+import java.util.function.Supplier;
 
-import static com.hazelcast.jet.impl.util.ExceptionUtil.rethrow;
-
-public class WritableSerializerHook {
+/**
+ * This class can be extended to quickly create strongly typed serializers for
+ * {@code Writable} types
+ */
+public abstract class WritableSerializerHook<T extends Writable> implements SerializerHook<T> {
 
-    private static final int WRITABLE = -400;
+    private final Class<T> clazz;
+    private final Supplier<T> supplier;
```

Comment:
`supplier` is a very generic name, I had to study all the code to figure out the purpose. Maybe `factory` is better, or `constructor` or something.